### PR TITLE
Allow passing options to lua profiles

### PR DESCRIPTION
**Pull request by karenzshea**
# Issue

Allows dynamic testing of profile changes by adding a lua function that will first look for a given variable in the run environment, and fallback to a given default.

## Tasklist
 - [x] Make regression / cucumber case pass
-  [ ] Add test for lua environment reading function
 - [ ] review
 - [ ] adjust for for comments

## Requirements / Relations
https://github.com/Project-OSRM/osrm-backend/issues/2781
https://github.com/Project-OSRM/osrm-backend/pull/2795

### CODE REVIEW

**Comment by karenzshea** [comment]
Going to wait on https://github.com/Project-OSRM/osrm-backend/pull/2795 before adding a new environment variable reading feature to our cucumber suite.

**Comment by MoKob** [comment]
I added the dependency of the cucumber refactor to the PR.

**Comment by daniel-j-h** [comment]
I guess we want to do this for all our flags and then add tests for both their true / false values.

File: profiles/car.lua [hunk 1]
```diff
@@ -158,7 +159,7 @@ local turn_bias                  = properties.left_hand_driving and 1/1.2 or 1.2
 local obey_oneway                = true
 local ignore_areas               = true
 local ignore_hov_ways            = true
-local ignore_toll_ways           = false
+local ignore_toll_ways = set_option("ignore_toll_ways", false)
```
Comment (daniel-j-h) [inline profiles/car.lua]: Looks like the `scripts/check-taginfo.py` does not like this ^ --- Travis failing.

**Comment by daniel-j-h** [comment]
Related: @karenzshea https://github.com/Project-OSRM/osrm-backend/pull/2954#issuecomment-250004546

**Comment by karenzshea** [comment]
Superseeded by https://github.com/Project-OSRM/osrm-backend/pull/2954!

**Pull request closed without merging**